src/mavlink/mavlink_camera.rs

```diff
@@ -607,15 +600,21 @@ impl MavlinkCamera {
                 });
                 if let Err(error) = sender.send(Message::ToBeSent((our_header, message))) {
                     warn!("Failed to send message: {error:?}");
+                    res = Err(error.into());
                 }
             });
+
+        res
     }
 }
 
 impl Drop for MavlinkCameraHandle {
     fn drop(&mut self) {
-        self.heartbeat_handle.abort();
-        self.messages_handle.abort();
-        super::manager::Manager::drop_id(self.inner.component.component_id)
+        if !self.heartbeat_handle.is_finished() || !self.messages_handle.is_finished() {
```

Comment:
It does not fix the issue compared with #314.
Change this if to a while and you'll see.

![image](https://github.com/mavlink/mavlink-camera-manager/assets/1215497/506e7d10-38f2-44ee-b53e-8631d194bd2c)

Reply:
Thanks. here that doesn't happen... Then I'm afraid I can't effectively validate the other as well.